portal-impl/src/com/liferay/portlet/blogs/service/impl/BlogsEntryLocalServiceImpl.java

```diff
@@ -684,10 +684,26 @@ public int getCompanyEntriesCount(
 
 		BlogsEntry entry = blogsEntryPersistence.findByPrimaryKey(entryId);
 
-		return blogsEntryPersistence.findByG_S_PrevAndNext(
-			entry.getEntryId(), entry.getGroupId(),
-			WorkflowConstants.STATUS_APPROVED,
-			new EntryDisplayDateComparator(true));
+		long groupId = entry.getGroupId();
+
+		Date displayDate = entry.getDisplayDate();
+
+		BlogsEntry[] blogsEntries =
+			blogsEntryPersistence.findByG_S_D_PrevAndNext(
+				entryId, groupId, WorkflowConstants.STATUS_APPROVED,
+				displayDate, new EntryDisplayDateComparator(true));
+
+		if (Validator.isNull(blogsEntries[0])) {
+			blogsEntries[0] = blogsEntryPersistence.fetchByG_LtD_S_First(
+				groupId, displayDate, WorkflowConstants.STATUS_APPROVED, null);
```

Comment:
Can we add a comparator here like `EntryDisplayDateComparator`. Since we are using fetch*First it's better if we provide a comparator.